src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.ts

```diff
@@ -26,9 +27,11 @@ const BASE_URL = 'rgw/bucket';
   styleUrls: ['./rgw-bucket-list.component.scss'],
   providers: [{ provide: URLBuilderService, useValue: new URLBuilderService(BASE_URL) }]
 })
-export class RgwBucketListComponent extends ListWithDetails {
+export class RgwBucketListComponent extends ListWithDetails implements OnInit {
   @ViewChild(TableComponent, { static: true })
   table: TableComponent;
+  @ViewChild('bucketUsageTpl', { static: true })
```

Comment:
Perhaps `static: false` (or nothing, as it's the default value for static) fix the change detector issue?

Reply:
No luck with that too!

Reply:
The problem is about a race condition when calling ```getBucketList()``` (synchronously) in template.
Also, the test is not testing the case when usage payload is a collection and you have to get aggregates, so to solve both
thing I'd suggest a refactor like this:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts
index b2b8238b3c..c595645958 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts
@@ -22,19 +22,7 @@ describe('RgwBucketListComponent', () => {
   let component: RgwBucketListComponent;
   let fixture: ComponentFixture<RgwBucketListComponent>;
   let rgwBucketService: RgwBucketService;
-
-  const getBucketList = (): any[] => {
-    return [  {
-      bucket: 'bucket',
-      owner: 'testid',
-      usage: {
-          'rgw.main': {
-              size_actual: 4,
-              size_utilized: 2,
-          }
-      }
-  }];
-  };
+  let rgwBucketServiceListSpy: jasmine.Spy;
 
   configureTestBed({
     declarations: [RgwBucketListComponent, RgwBucketDetailsComponent],
@@ -50,6 +38,9 @@ describe('RgwBucketListComponent', () => {
   });
 
   beforeEach(() => {
+    rgwBucketService = TestBed.get(RgwBucketService);
+    rgwBucketServiceListSpy = spyOn(rgwBucketService, 'list');
+    rgwBucketServiceListSpy.and.returnValue(of(null));
     fixture = TestBed.createComponent(RgwBucketListComponent);
     component = fixture.componentInstance;
   });
@@ -102,9 +93,36 @@ describe('RgwBucketListComponent', () => {
   });
 
   it('should test if bucket data is tranformed correctly', () => {
-    rgwBucketService = TestBed.get(RgwBucketService);
-    spyOn(rgwBucketService, 'list').and.callFake(() => of(getBucketList()));
+    rgwBucketServiceListSpy.and.returnValue(
+      of([
+        {
+          bucket: 'bucket',
+          owner: 'testid',
+          usage: {
+            'rgw.main': {
+              size_actual: 4,
+              size_utilized: 2
+            },
+            'rgw.another': {
+              size_actual: 6,
+              size_utilized: 6
+            }
+          }
+        }
+      ])
+    );
     fixture.detectChanges();
-    expect(component.buckets).toEqual([{bucket: 'bucket', owner: 'testid', usage: {'rgw.main': {size_actual: 4, size_utilized: 2}}, usageprop: 0.5}]);
+
+    expect(component.buckets).toEqual([
+      {
+        bucket: 'bucket',
+        owner: 'testid',
+        usage: {
+          'rgw.main': { size_actual: 4, size_utilized: 2 },
+          'rgw.another': { size_actual: 6, size_utilized: 6 }
+        },
+        usageprop: 0.8
+      }
+    ]);
   });
 });
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.ts
index 503bbc782a..ab062a8ee1 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.ts
@@ -1,4 +1,12 @@
-import { Component, NgZone, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import {
+  AfterContentChecked,
+  ChangeDetectorRef,
+  Component,
+  NgZone,
+  OnInit,
+  TemplateRef,
+  ViewChild
+} from '@angular/core';
 
 import { I18n } from '@ngx-translate/i18n-polyfill';
 import * as _ from 'lodash';
@@ -27,7 +35,7 @@ const BASE_URL = 'rgw/bucket';
   styleUrls: ['./rgw-bucket-list.component.scss'],
   providers: [{ provide: URLBuilderService, useValue: new URLBuilderService(BASE_URL) }]
 })
-export class RgwBucketListComponent extends ListWithDetails implements OnInit {
+export class RgwBucketListComponent extends ListWithDetails implements OnInit, AfterContentChecked {
   @ViewChild(TableComponent, { static: true })
   table: TableComponent;
   @ViewChild('bucketUsageTpl', { static: true })
@@ -48,7 +56,8 @@ export class RgwBucketListComponent extends ListWithDetails implements OnInit {
     private i18n: I18n,
     private urlBuilder: URLBuilderService,
     public actionLabels: ActionLabelsI18n,
-    private ngZone: NgZone
+    private ngZone: NgZone,
+    private changeDetectorRef: ChangeDetectorRef
   ) {
     super();
     this.permission = this.authStorageService.getPermissions().rgw;
@@ -100,15 +109,19 @@ export class RgwBucketListComponent extends ListWithDetails implements OnInit {
     ];
   }
 
-  transformsbucketData() {
+  ngAfterContentChecked(): void {
+    this.changeDetectorRef.detectChanges();
+  }
+
+  transformBucketsData() {
     _.forEach(this.buckets, (bucketKey) => {
       const usageList = bucketKey['usage'];
       let utilizedBucketSize = 0;
       let actualBucketSize = 0;
-        _.forEach(usageList, (usageKey) => {
-          utilizedBucketSize = utilizedBucketSize + usageKey.size_utilized;
-          actualBucketSize = actualBucketSize + usageKey.size_actual;
-        });
+      _.forEach(usageList, (usageKey) => {
+        utilizedBucketSize = utilizedBucketSize + usageKey.size_utilized;
+        actualBucketSize = actualBucketSize + usageKey.size_actual;
+      });
       bucketKey['usageprop'] =
         actualBucketSize > 0 ? utilizedBucketSize / actualBucketSize : actualBucketSize;
     });
@@ -131,7 +144,7 @@ export class RgwBucketListComponent extends ListWithDetails implements OnInit {
     this.rgwBucketService.list().subscribe(
       (resp: object[]) => {
         this.buckets = resp;
-        this.transformsbucketData();
+        this.transformBucketsData();
       },
       () => {
         context.error();
```